File: core/core-domain/src/main/java/com/coffee/gu/menu/Price.java
package com.coffee.gu.menu;



import com.coffee.gu.CoreException;
import com.coffee.gu.ErrorType;

import java.math.BigDecimal;

public record Price(
        BigDecimal costPrice,
        BigDecimal salesPrice
) {
    public Price {
        if (salesPrice.compareTo(costPrice) < 0) {
            throw new CoreException(ErrorType.INVALID_REQUEST, "판매가는 정가보다 낮을 수 없습니다.");
        }
    }

}

File: core/core-domain/src/main/kotlin/com/coffee/gu/menu/Price.kt
package com.coffee.gu.menu

import com.coffee.gu.CoreException
import com.coffee.gu.ErrorType
import java.math.BigDecimal

class Price(
    val costPrice: BigDecimal,
    val salesPrice: BigDecimal,
) {
    init {
        if (salesPrice < costPrice) {
            throw CoreException(ErrorType.INVALID_REQUEST, "판매가는 정가보다 낮을 수 없습니다.")
        }
    }
}
